lib/msf/ui/console/command_dispatcher/creds.rb

```diff
@@ -490,86 +484,20 @@ def creds_search(*args)
-          # If none of this Core's associated Logins is for a host within
-          # the user-supplied RangeWalker, then we don't have any reason to
-          # print it out. However, we treat the absence of ranges as meaning
-          # all hosts.
-          if host_ranges.present? && !host_ranges.any? { |range| range.include?(service.host.address) }
-            next
-          end
-
-          row = [ service.host.address ]
-          row << origin
-          rhosts << service.host.address
-          row << build_service_info(service)
-
-          matched_cred_ids << core.id
-          public_val = core.public ? core.public.username : ""
-          private_val = core.private ? core.private.to_s : ""
-          if truncate && private_val.to_s.length > 87
-            private_val = "#{private_val[0,87]} (TRUNCATED)"
-          end
-          realm_val = core.realm ? core.realm.value : ""
-          human_val = core.private ? core.private.class.model_name.human : ""
-          if human_val == ""
-            jtr_val = "" #11433, private can be nil
-          else
-            jtr_val = core.private.jtr_format ? core.private.jtr_format : ""
-          end
-
-          row += [
-            public_val,
-            private_val,
-            realm_val,
-            human_val,
-            jtr_val
-          ]
-          tbl << row
-        end
       end
     end
-    if mode == :delete
-      result = framework.db.delete_credentials(ids: matched_cred_ids)
-      delete_count = result.size
-    end
 
     if output_file.nil?
-      print_line(tbl.to_s)
+      print_line(tbl.to_s) if tbl
     else
-      if output_file.end_with? '.jtr'
-        hashlist = ::File.open(output_file, "wb")
-        ['Metasploit::Credential::NonreplayableHash',
-         'Metasploit::Credential::PostgresMD5',
-         'Metasploit::Credential::NTLMHash'].each do |type|
-          framework.db.creds(type: type).each do |core|
-            formatted = hash_to_jtr(core)
-            unless formatted.nil?
-              hashlist.puts formatted
-            end
-          end
-        end
-        hashlist.close
-      elsif output_file.end_with? '.hcat'
-        hashlist = ::File.open(output_file, "wb")
-        ['Metasploit::Credential::NonreplayableHash',
-         'Metasploit::Credential::PostgresMD5',
-         'Metasploit::Credential::NTLMHash'].each do |type|
-          framework.db.creds(type: type).each do |core|
-            formatted = hash_to_hashcat(core)
-            unless formatted.nil?
-              hashlist.puts formatted
-            end
-          end
-        end
-        hashlist.close
-      else #csv
-        # create the output file
-        ::File.open(output_file, "wb") { |f| f.write(tbl.to_csv) }
-      end
-      print_status("Wrote creds to #{output_file}")
+      output_file.write(tbl.to_csv) if output_formatter.nil? && tbl
+      output_file.close
+      print_status("Wrote creds to #{output_file.path}")
```

Comment:
So if both `output_formatter` is nil and `tbl` is nil, then you will still print out a message here saying that you wrote creds to a file even though this didn't occur. This doesn't seem like a good outcome :/

Reply:
The condition you're describing shouldn't be recheable with any combination of options. Same thing with your other comment. If someone is editing this file though they'd probably want to see the exception and stack trace describing what went wrong (`tbl` wasn't initialized). I'll update this and the other occurrence to remove the check since it'll always be set.